cmd/kubeadm/app/cmd/alpha/certs.go

```diff
@@ -245,13 +261,37 @@ func renewCert(flags *renewFlags, kdir string, handler *renewal.CertificateRenew
 	return nil
 }
 
+func getInternalCfg(cfgPath string, kubeconfigPath string, cfg kubeadmapiv1beta2.ClusterConfiguration, out io.Writer, logPrefix string) (*kubeadmapi.InitConfiguration, error) {
+	// In case the user is not providing a custom config, try to get current config from the cluster.
+	// NB. this operation should not block, because we want to provide allow certificate renewal also in case of not working clusters
+	if cfgPath == "" {
+		client, err := kubeconfigutil.ClientSetFromFile(kubeconfigPath)
+		if err == nil {
+			internalcfg, err := configutil.FetchInitConfigurationFromCluster(client, out, logPrefix, false)
```

Comment:
We may end up using defaulted config in case we cannot connect to an API server or something else happens here.

Reply:
yes, this is intentional.
The command works at "best-effort"; in case something goes wrong the user gets a warning but the command goes on